sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -317,7 +318,19 @@ def increment(self, settings, response=None, error=None):
                 settings['status'] -= 1
                 settings['history'].append(RequestHistory(response.http_request, http_response=response.http_response))
 
-        return not self.is_exhausted(settings)
+        if not self.is_exhausted(settings):
+            if response.http_request.body and hasattr(response.http_request.body, 'read'):
```

Comment:
"body" is bytes, so body can't have "read()"
https://github.com/Azure/azure-sdk-for-python/blob/18bec9767052ffabd9cf4a4ac38ea047f190a442/sdk/core/azure-core/azure/core/pipeline/transport/_base.py#L235-L239